tests/test_pydantic_io_syntax.py

```diff
@@ -0,0 +1,62 @@
+import pytest
+
+from hera.shared._global_config import _SCRIPT_PYDANTIC_IO_FLAG
+from hera.workflows import Input, Output, Steps, Workflow, script
+
+
+class IntInput(Input):
+    field: int
+
+
+class IntOutput(Output):
+    field: int
+
+
+@pytest.fixture(autouse=True)
+def enable_pydantic_io(global_config_fixture):
+    global_config_fixture.experimental_features[_SCRIPT_PYDANTIC_IO_FLAG] = True
+
+
+def test_output_field_contains_argo_template(global_config_fixture):
+    @script()
+    def triple(input: IntInput) -> IntOutput:
+        return IntOutput(field=input.field * 3)
+
+    with Workflow(name="foo"):
+        with Steps(name="bar"):
+            result = triple(IntInput(field=5)).field
+
+    assert result == "{{steps.triple.outputs.parameters.field}}"
```

Comment:
This is not what the Pydantic IO feature flag was designed for and shouldn't be possible in a normal `Steps` context

Reply:
Moved behind the decorator_syntax flag. I think it's important to incremental migration to allow this in a `Steps` context though.